File: go/grind75/SameTree.go
package grind75

func isSameTree(left *TreeNode, right *TreeNode) bool {
	if left == nil || right == nil {
		return left == right
	}
	if left.Val == right.Val {
		return isSameTree(left.Left, right.Left) && isSameTree(left.Right, right.Right)
	}
	return false
}

File: src/main/java/blind75/SameTree.java
package blind75;

public class SameTree {
    public boolean isSameTree(TreeNode left, TreeNode right) {
        if (left == null || right == null) return (left == right);
        if (left.val == right.val) return isSameTree(left.left, right.left) && isSameTree(left.right, right.right);
        return false;
    }
}
